sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/_patch.py

```diff
@@ -75,6 +78,17 @@ def __init__(
         ledger_certificate_path: Union[bytes, str, os.PathLike],
         **kwargs: Any,
     ) -> None:
+        # Remove some kwargs first so that there aren't unexpected kwargs passed to
+        # get_ledger_identity.
+        if isinstance(credential, ConfidentialLedgerCertificateCredential):
```

Comment:
You can re-org your code and combine

two sections of `if isinstance(credential, ConfidentialLedgerCertificateCredential): ` together.

Reply:
Unfortunately, we can't. The second `if` does `kwargs["connection_cert"] = credential.certificate_path`, which adds a kwarg that is only relevant for the constructor - its presence would cause line 98 (`ledger_cert = identity_service_client.get_ledger_identity(ledger_id, **kwargs)`) to fail.

We also have to pop `authentication_policy` in the `else` section of the first `if`, as its presence would also cause an error on line 98.

Reply:
You don't have to put connection_cert into kwargs.

Just add connection_cert=credential.certificate_path when you need it.

Reply:
I don't know if that's any better. Without using kwargs, line 117 becomes
```python
if isinstance(credential, ConfidentialLedgerCertificateCredential):
    super().__init__(endpoint, connection_cert=credential.certificate_path, **kwargs)
else:
    super().__init__(endpoint, **kwargs)
```
or possibly
```python
super().__init__(endpoint, connection_cert=credential.certificate_path if isinstance(credential, ConfidentialLedgerCertificateCredential) else None, **kwargs)
```

which is still two `if isinstance(credential, ConfidentialLedgerCertificateCredential)` statements.